### PR TITLE
add default values for kubeapiserverconfig to make it easier to use

### PR DESCRIPTION
**Pull request by deads2k**
Just the last commit is unique.  This adds value defaulting for the kubeapiserverconfig.

I plan to add some logic cluster up to prime the configmaps and secrets with content that currently exists in the local folder.

/assign @sttts @mfojtik

### CODE REVIEW

File: pkg/oc/clusterup/run_self_hosted.go [hunk 1]
```diff
@@ -68,8 +68,18 @@ var (
...
+	runlevelZeroLabel         = map[string]string{"openshift.io/run-level": "0"}
 	runlevelOneLabel          = map[string]string{"openshift.io/run-level": "1"}
 	runLevelOneKubeComponents = []componentInstallTemplate{
+		{
+			ComponentImage: "hypershift",
+			Template: componentinstall.Template{
+				Name:            "etcd",
+				Namespace:       "kube-system",
+				NamespaceObj:    newNamespaceBytes("kube-system", runlevelZeroLabel),
+				InstallTemplate: manifests.MustAsset("install/etcd/install.yaml"),
```
Comment (mfojtik) [inline pkg/oc/clusterup/run_self_hosted.go]: hrm, I don't understand this (and seems like this is not related to the commit), but is this just going to install a service for etcd ?

File: install/etcd/install.yaml [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+  kind: Service
+  metadata:
+    name: etcd
+    namespace: ${NAMESPACE}
+  spec:
+    selector:
+      openshift.io/component: etcd
+    ports:
+    - name: listen
+      port: 4001
```
Comment (mfojtik) [inline install/etcd/install.yaml]: 2379 ?

File: pkg/cmd/openshift-kube-apiserver/configdefault/config_default.go [hunk 3]
```diff
@@ -0,0 +1,54 @@
+package configdefault
+
+import (
+	configv1 "github.com/openshift/api/config/v1"
+	"github.com/openshift/library-go/pkg/crypto"
+)
+
+func DefaultString(target *string, defaultVal string) {
+	if len(*target) == 0 {
```
Comment (mfojtik) [inline pkg/cmd/openshift-kube-apiserver/configdefault/config_default.go]: should you check for nil here?

File: pkg/cmd/openshift-kube-apiserver/configdefault/config_default.go [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+		*target = defaultVal
+	}
+}
+
+func DefaultStringSlice(target *[]string, defaultVal []string) {
+	if len(*target) == 0 {
```
Comment (mfojtik) [inline pkg/cmd/openshift-kube-apiserver/configdefault/config_default.go]: nil check?

File: install/etcd/install.yaml (hunk 2 above)
Comment (sttts) [reply install/etcd/install.yaml]: and reason not to switch? 4001 is like 3+ years old.

File: pkg/cmd/openshift-kube-apiserver/configdefault/config_default.go (hunk 3 above)
Comment (sttts) [reply pkg/cmd/openshift-kube-apiserver/configdefault/config_default.go]: not necessary IMO. Non-nil is a programming error as this is merely a reference to a static fields. It should panic.

**Comment by sttts** [comment]
last commit lgtm.

File: install/etcd/install.yaml (hunk 2 above)
Comment (deads2k) [reply install/etcd/install.yaml]: > and reason not to switch? 4001 is like 3+ years old.

doesn't currently work that way?

**Pull request merged**